configuration-service/handlers/project.go

```diff
@@ -120,6 +120,10 @@ func PutProjectProjectNameHandlerFunc(params project.PutProjectProjectNameParams
 					logger.Error("Authentication error detected")
 					return project.NewPostProjectDefault(http.StatusFailedDependency).WithPayload(&models.Error{Code: http.StatusFailedDependency, Message: swag.String(err.Error())})
 				}
+				if strings.Contains(err.Error(), "failed to set upstream") {
```

Comment:
Is the "failed to set upstream" message always an indicator for a non-existing upstream URL? Otherwise we might need to adapt the `UpdateOrCreateOrigin` function to return proper errors for different failure scenarios (e.g. unknown repo URL, authentication errors, etc)

Reply:
upstream not found covers case of unexisting URL and wrong token, however I just tested that in case of wrong user no error is returned!!!!

Reply:
In any case, please do not make decisions based on log or error messages. It's just too easy to break without knowing.